components/ntp_background_images/browser/ntp_background_images_service.cc

```diff
@@ -131,20 +135,33 @@ void NTPBackgroundImagesService::Init() {
       DVLOG(2) << __func__ << ": NTP SR test data will be loaded"
                << " from local path at: "
                << forced_local_path.LossyDisplayName();
-      OnComponentReady(false, forced_local_path);
+      OnSponsoredComponentReady(false, forced_local_path);
     } else {
       CheckSuperReferralComponent();
     }
   }
 #endif
 }
 
-void NTPBackgroundImagesService::CheckSIComponentUpdate(
+void NTPBackgroundImagesService::CheckImagesComponentUpdate(
     const std::string& component_id) {
-  DVLOG(2) << __func__ << ": Check NTP SI component update";
+  DVLOG(2) << __func__ << ": Check NTP Images component update";
   BraveOnDemandUpdater::GetInstance()->OnDemandUpdate(component_id);
 }
 
+void NTPBackgroundImagesService::RegisterBackgroundImagesComponent() {
+  const auto& data = GetBackgroundImagesComponentData();
+
+  DVLOG(2) << __func__ << ": Start NTP BI component";
+  RegisterNTPBackgroundImagesComponent(
```

Comment:
PTAL my previous comment - https://github.com/brave/brave-core/pull/9635/files#r682247401

Reply:
What I meant was calling like below.
We don't need to pass NTPBI specifi data as arguments.
`RegisterNTPBackgroundImagesComponent()` already knows about it.
```
RegisterNTPBackgroundImagesComponent(
    component_update_service_,
    base::BindRepeating(&NTPBackgroundImagesService::OnComponentReady,
                        weak_factory_.GetWeakPtr()));
```